### PR TITLE
Feature/transfers

### PR DESCRIPTION
**Pull request by OgnjenCBS**

### CODE REVIEW

File: src/modules/transactions/transactions.service.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { PageOptionsDto } from 'src/utils/pagination/page-options.dto';
+import { PageDto } from 'src/utils/pagination/page.dto';
+import { TransactionDto } from './dto/transactions.dto';
+import { TransactionsRepository } from './transactions.repository';
+
+@Injectable()
+export class TransactionsService {
+  constructor(private readonly txRepo: TransactionsRepository) {}
+
+  public async transaction(
```
Comment (Str4LeGit) [inline src/modules/transactions/transactions.service.ts:11]:
```suggestion
  public async getTransactionsByAccountId(
```
Or similar

**Review by Str4LeGit — APPROVED** [review]
👍 one minor comment

File: src/database/mysql/migrations/1700669840501-create-transactions-table.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+{
+  name = 'createTransactionsTable1700669840501';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`CREATE TABLE \`Transactions\` (
+            \`id\` int NOT NULL AUTO_INCREMENT, 
+            \`sender_account_id\` varchar(255) NOT NULL, 
+            \`asset_id\` varchar(255) NOT NULL, 
+            \`amount\` float NOT NULL, 
+            \`receiver_acc_id\` varchar(255) NOT NULL,
```
Comment (DanijelCBS) [inline src/database/mysql/migrations/1700669840501-create-transactions-table.ts:14]: Name this receiver_account_id

File: src/modules/transactions/entity/transactions.entity.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  @Column({ name: 'sender_account_id' })
+  senderAccId: string;
+
+  @Column({ name: 'asset_id' })
+  asset: string;
+
+  @Column('float', { name: 'amount' })
+  amount: number;
+
+  @Column({ name: 'receiver_acc_id' })
```
Comment (DanijelCBS) [inline src/modules/transactions/entity/transactions.entity.ts:17]: Change it here as well

File: src/modules/transactions/entity/transactions.entity.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity(`Transactions`)
+export class Transaction {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column({ name: 'sender_account_id' })
+  senderAccId: string;
```
Comment (DanijelCBS) [inline src/modules/transactions/entity/transactions.entity.ts:9]: Name it senderAccountId

File: src/modules/transactions/entity/transactions.entity.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+  @Column({ name: 'asset_id' })
+  asset: string;
+
+  @Column('float', { name: 'amount' })
+  amount: number;
+
+  @Column({ name: 'receiver_acc_id' })
+  receiverAccId: string;
```
Comment (DanijelCBS) [inline src/modules/transactions/entity/transactions.entity.ts:18]: Name it receiverAccountId

File: src/modules/transactions/mapper/tx-entity-to-dto.mapper.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { BaseDtoMapper } from 'src/utils/mappers/base-dto-mapper';
+import { Transaction } from '../entity/transactions.entity';
+import { TransactionDto } from '../dto/transactions.dto';
+
```
Comment (DanijelCBS) [inline src/modules/transactions/mapper/tx-entity-to-dto.mapper.ts:4]: Change naming here as well

File: src/modules/transactions/transactions.repository.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+  public async findTxsByAccountId(
+    pageOptions: PageOptionsDto,
+    accountId: string,
+  ): Promise<PageDto<TransactionDto>> {
+    const [data, count] = await this.txRepository.findAndCount({
+      skip: pageOptions.skip,
+      take: pageOptions.size,
+      order: { id: 'DESC' },
+      where: [{ senderAccId: accountId }, { receiverAccId: accountId }],
```
Comment (DanijelCBS) [inline src/modules/transactions/transactions.repository.ts:31]: Here as well

**Review by DanijelCBS — CHANGES_REQUESTED** [review]

File: src/modules/transfers/dto/transfer.dto.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+export interface TransferDto {
+  sender: string;
+  amount: number;
+  asset: string;
+  receiver: string;
+  transferedAt: Date;
```
Comment (DanijelCBS) [inline src/modules/transfers/dto/transfer.dto.ts:6]: Typo, it should be transferredAt

**Review by DanijelCBS — APPROVED** [review]

**Review by DanijelCBS — APPROVED** [review]

**Pull request merged**